src/data/ulift.lean

```diff
@@ -1,17 +1,104 @@
 /-
 Copyright (c) 2018 Scott Morrison. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Scott Morrison
+Authors: Scott Morrison, Jannis Limperg
 
-Transport through constant families.
+Facts about `ulift` and `plift`.
 -/
 
-universes u₁ u₂
+universes u v
 
-@[simp] lemma plift.rec.constant {α : Sort u₁} {β : Sort u₂} (b : β) :
+namespace plift
+
+variables {α β : Sort u}
+
+/-- Functorial action. -/
+@[simp] def map (f : α → β) : plift α → plift β
+| (up a) := up (f a)
+
+/-- Embedding of pure values. -/
+@[simp] def pure : α → plift α := up
```

Comment:
What universe is it being lifted to here? Won't it just be `u` again by default without any annotations?

Reply:
`plift.{u} : Sort u → Type u` only has a single universe parameter.

Reply:
Good question Chris, thank you. It seems like Lean does the right thing by default and generalises the universes as much as possible for the `ulift` definitions. (For `plift` it has no choice, as Gabriel points out.) However, I've found a way to make `ulift.bind` even more universe polymorphic.